provisioning/device/src/Models/DeviceRegistrationResult.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client
+{
+    /// <summary>
+    /// The result of a registration operation.
+    /// </summary>
+    public class DeviceRegistrationResult
+    {
+        /// <summary>
+        /// For serialization.
+        /// </summary>
+        internal DeviceRegistrationResult()
+        {
+
+        }
+
+        /// <summary>
+        ///. Creates an instance of this class.
+        /// </summary>
+        public DeviceRegistrationResult(
```

Comment:
iirc only one of these constructors is actually being used by the SDK but we couldn't remove the unused ones because they were public. We can remove the unused ones and change the visibility to internal.